### PR TITLE
Add note to README to use Rust nightly

### PR DESCRIPTION
**Pull request by spiderkeys**
The current build instructions fail if using the default stable release of Rust installed by rustup due to the use of some unstable features:
```
error[E0554]: `#![feature]` may not be used on the stable release channel
  --> /home/spiderkeys/.cargo/git/checkouts/zenoh-cc237f2570fab813/12bfe92/zenoh-router/src/lib.rs:14:1
   |
14 | #![feature(get_mut_unchecked)]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0554]: `#![feature]` may not be used on the stable release channel
  --> /home/spiderkeys/.cargo/git/checkouts/zenoh-cc237f2570fab813/12bfe92/zenoh-router/src/lib.rs:15:1
   |
15 | #![feature(async_closure)]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0554]: `#![feature]` may not be used on the stable release channel
  --> /home/spiderkeys/.cargo/git/checkouts/zenoh-cc237f2570fab813/12bfe92/zenoh-router/src/lib.rs:16:1
   |
16 | #![feature(map_into_keys_values)]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to 3 previous errors

For more information about this error, try `rustc --explain E0554`.
error: could not compile `zenoh-router`

To learn more, run the command again with --verbose.
warning: build failed, waiting for other jobs to finish...
error: build failed
```
This PR updates the README to include instructions for installing nightly and using it to build.

### CODE REVIEW

**Comment by kydos** [comment]
@spiderkeys  thanks for your contribution, to be able to merge it you need to do the Eclipse Contributor Agreement (see https://www.eclipse.org/legal/ECA.php). Once that is in order I'll be able to merge your pull request.

**Comment by spiderkeys** [comment]
Thanks @kydos, I've just done the ECA. Let me know if there are any other issues.

**Comment by kydos** [comment]
Great, I'll merge once I get the green light.

**Comment by kydos** [comment]
BTW, do not hesitate to reach out if you have any questions. If you are going to do some tests and want to use the resource generalisation, that requires a configuration that we have not pushed yet. Finally, do not hesitate to post your questions on http://gitter.im/atolab/zenoh

Thanks again for your contribution!

**Review by kydos — APPROVED** [review]

**Pull request merged**
